### ISSUE TITLE
Include `None` in signatures with an implicit default `None`

### ISSUE THREAD

**Issue by RobertCraigie** [microsoft/pyright#3458]
**Is your feature request related to a problem? Please describe.**

I'm writing a class that optionally supports forwarding some arguments to `httpx`'s `Client` class. As `httpx` uses mypy's implicit `None` feature, their type annotations are missing that detail (https://github.com/encode/httpx/blob/6f31bc4bea7bbd03260af6c226b24c541a203130/httpx/_client.py#L156-L168). 

For example:

```py
import httpx

class Foo:
    def __init__(self, proxies: httpx.Proxy | None = None) -> None:
        self._client = httpx.Client(proxies=proxies)  # error: Type "None" cannot be assigned to type "ProxiesTypes"
```

This is actually fine at runtime but causes an error because of the incorrect type annotation on the `httpx` side.

**Describe the solution you'd like**

I appreciate this is an error on the part of the library author but Ideally pyright could `Union` the given type annotation with the type of the default value.

**Comment by erictraut** [comment]
You can set `strictParameterNoneValue` to false if you want pyright to interpret a default value of `None` as an implied `Optional` for parameters. This was the original way PEP 484 was written, but more recently (about two years ago), PEP 484 was updated with a clarification saying that type checkers should require an explicit `None` union type or an `Optional`. I recommend reporting the bug to the library maintainers so they update their code accordingly.
